cg/apps/cgstats/crud/create.py

```diff
@@ -142,48 +172,124 @@ def create_projects(manager: StatsAPI, project_names: Iterable[str]) -> Dict[str
     return project_name_to_id
 
 
-def create_samples(
+def _create_samples(
+    manager: StatsAPI, sample: NovaSeqSample, project_name_to_id: Dict[str, int]
+) -> Union[int, None]:
+    """handles sample objects creation in the table `Sample` in cgstats"""
+
+    barcode = (
+        sample.index if not sample.second_index else "+".join([sample.index, sample.second_index])
+    )
+    sample_id: Optional[int] = find.get_sample_id(sample_id=sample.sample_id, barcode=barcode)
+    if sample.project == "indexcheck":
+        LOG.debug("Skip adding indexcheck sample to database")
+        return
+    project_id: int = project_name_to_id[sample.project]
+    if not sample_id:
+        sample_object: stats_models.Sample = create_sample(
+            manager=manager,
+            sample_id=sample.sample_id,
+            barcode=barcode,
+            project_id=project_id,
+        )
+        sample_id: int = sample_object.sample_id
```

Comment:
That's not possible in this case, because `sample_id` is generated by `create_sample` above. In this case `sample_id` is the primary key of the table `Sample` in cgstats.